diff --git a/chat/public/js/components/chat_add_room.js b/chat/public/js/components/chat_add_room.js
index dd6aec0..2be393d 100644
--- a/chat/public/js/components/chat_add_room.js
+++ b/chat/public/js/components/chat_add_room.js
@@ -19,7 +19,7 @@ export default class ChatAddRoom {
           label: __('Room Type'),
           fieldname: 'type',
           fieldtype: 'Select',
-          options: ['Group', 'Direct'],
+          options: ['Group'],
           default: 'Group',
           onchange: () => {
             const type = this.add_room_dialog.get_value('type');
@@ -33,7 +33,8 @@ export default class ChatAddRoom {
         {
           label: __('Room Name'),
           fieldname: 'room_name',
-          fieldtype: 'Data',
+          fieldtype: 'Select',
+          options:job_order_name(),
           depends_on: "eval:doc.type == 'Group'",
           reqd: true,
         },
@@ -42,8 +43,9 @@ export default class ChatAddRoom {
           fieldname: 'users',
           fieldtype: 'MultiSelectPills',
           options: this.users_list,
+          default:this.users,
           depends_on: "eval:doc.type == 'Group'",
-          reqd: true,
+          
         },
         {
           label: __('User'),
@@ -57,13 +59,13 @@ export default class ChatAddRoom {
         primary: {
           label: __('Create'),
           onsubmit: (values) => {
-            let users = this.add_room_dialog.fields_dict.users.get_values();
+            var users = this.add_room_dialog.fields_dict.users.get_values();
             let room_name = values.room_name;
             if (values.type === 'Direct') {
               users = [values.user];
               room_name = 'Direct Room';
             }
-            this.handle_room_creation(room_name, users, values.type);
+            this.handle_room_creation(room_name, users=users, values.type);
             this.add_room_dialog.hide();
           },
         },
@@ -84,3 +86,19 @@ export default class ChatAddRoom {
     }
   }
 }
+
+function job_order_name(){
+  var user=frappe.session.user
+  var company_type = frappe.boot.tag.tag_user_info.company_type
+  var company_name = frappe.boot.tag.tag_user_info.company
+  let company = '';
+	frappe.call({
+		"method": "tag_workflow.chat_tag_data.job_order_name",
+		"args": {"username": user,"company_type":company_type,"company_name":company_name},
+		"async": 0,
+		"callback": function(r){
+			company += r.message;
+		}
+	});
+	return company
+}
