diff --git a/frappe/public/js/frappe/form/controls/date.js b/frappe/public/js/frappe/form/controls/date.js
index 5a4f521..d863562 100644
--- a/frappe/public/js/frappe/form/controls/date.js
+++ b/frappe/public/js/frappe/form/controls/date.js
@@ -68,7 +68,13 @@ frappe.ui.form.ControlDate = class ControlDate extends frappe.ui.form.ControlDat
 			maxDate: this.df.max_date,
 			firstDay: frappe.datetime.get_first_day_of_the_week_index(),
 			onSelect: () => {
-				this.$input.trigger("change");
+				$('.datepicker').on("mouseup",()=>{
+					this.$input.trigger("change");
+				});
+
+				$('.datepicker').on("click",()=>{
+					this.$input.trigger("change");
+				});
 			},
 			onShow: () => {
 				this.datepicker.$datepicker
