diff --git a/frappe/public/js/frappe/form/form.js b/frappe/public/js/frappe/form/form.js
index a8c73ba..e8bae9a 100644
--- a/frappe/public/js/frappe/form/form.js
+++ b/frappe/public/js/frappe/form/form.js
@@ -45,8 +45,8 @@ frappe.ui.form.Form = class FrappeForm {
 		this.beforeUnloadListener = (event) => {
 			event.preventDefault();
 			// A String is returned for compatability with older Browsers. Return Value has to be truthy to trigger "Leave Site" Dialog
-			return (event.returnValue =
-				"There are unsaved changes, are you sure you want to exit?");
+			// return (event.returnValue =
+			// 	"There are unsaved changes, are you sure you want to exit?");
 		};
 	}
 
@@ -376,7 +376,7 @@ frappe.ui.form.Form = class FrappeForm {
 	refresh(docname) {
 		var switched = docname ? true : false;
 
-		removeEventListener("beforeunload", this.beforeUnloadListener, { capture: true });
+		// removeEventListener("beforeunload", this.beforeUnloadListener, { capture: true });
 
 		if (docname) {
 			this.switch_doc(docname);
@@ -1338,9 +1338,9 @@ frappe.ui.form.Form = class FrappeForm {
 	dirty() {
 		this.doc.__unsaved = 1;
 		this.$wrapper.trigger("dirty");
-		if (!frappe.boot.developer_mode) {
-			addEventListener("beforeunload", this.beforeUnloadListener, { capture: true });
-		}
+		// if (!frappe.boot.developer_mode) {
+		// 	addEventListener("beforeunload", this.beforeUnloadListener, { capture: true });
+		// }
 	}
 
 	get_docinfo() {
